src/sections/Section4.jsx

```diff
@@ -39,6 +62,43 @@ function RpcInformation(){
         }
     };
 
+    const switchChain = useCallback(
+    async () => {
+      if (!window.ethereum) return;
+      const rawEthereumProvider = window.ethereum;
+      let { chainId, name, originalUrl, coin} = chains.filter(chain => chain.value === rpc)[0];
+      console.log(chainId)
+      console.log(originalUrl)
+      const chainIdHex = hexValue(parseInt(chainId));
+      try {
+        await rawEthereumProvider.request({
+          method: 'wallet_switchEthereumChain',
+          params: [{ chainId: chainIdHex }]
+        })
+      } catch (error) {
+        try {
+          if (error.code === 4902) {
+            await rawEthereumProvider.request({
+              method: 'wallet_addEthereumChain',
+              params: [{
+                chainId: chainIdHex,
+                chainName: name,
```

Comment:
Please use exactly the same data that you can see under the RPC dropdown when switching.

Reply:
The same network data from the current state of the component is now used to add the new network.
There was a small bug with the tab size when formatting the document in Visual Studio, the original tab size was restored.